mongo/options/encryptoptions.go

```diff
@@ -10,11 +10,21 @@ import (
 	"go.mongodb.org/mongo-driver/bson/primitive"
 )
 
+// QueryType describes the type of query the result of Encrypt is used for.
+type QueryType int
```

Comment:
What are your thoughts on using `uint8` here?  `byte` is an alias for `uint8`.

Reply:
Good question. I am not sure what is a more common pattern. The QueryType will have the same values as the [C enum type in libmongocrypt](https://github.com/mongodb/libmongocrypt/blob/master/src/mongocrypt.h.in#L1506).  A C enum is guaranteed to be the size of an int. I think `int` may be appropriate.